### ISSUE TITLE
Infer bad type

### ISSUE THREAD

**Issue by Transrian** [frictionlessdata/tableschema-py#209]
Hi guys,

I just discovered this project, and I met one strange problem while reading a CSV file, hope you can help about that :

Python code :

```python
#!/bin/env python3

from tabulate import tabulate
from tableschema import Table


table = Table("bug.csv")

table.infer()

print(tabulate(table.schema.descriptor['fields'], headers="keys", tablefmt="pipe"))

try:
    table.read(keyed=False, cast=True)
except Exception as err:
    print(err.errors)
    exit(1)
```

bug.csv(v1):
```csv
id,lenght
1,456
2,456.2
```
bug.csv(v2):
```csv
id,lenght
1,456
2,456.2
3,456
```

When using the v1 file, result are as expected, wereas when using the v2 file, something strange happend, please see..

v1:
```bash
$ python3 bug.py 
| name   | type    | format   |
|:-------|:--------|:---------|
| id     | integer | default  |
| lenght | number  | default  |
```

v2:
```bash
$ python3 bug.py 
| name   | type    | format   |
|:-------|:--------|:---------|
| id     | integer | default  |
| lenght | integer | default  |

[CastError('Field "lenght" can\'t cast value "456.2" for type "integer" with format "default"',)]
```

There are, for me, two problems :
- first, why is the first working whereas the second is not ? (why number type became integer ?)
- and then, wouldn't it be easier, for the sake of the initial data, to take the more 'permissive' type, here integer, instead of those that seems represent the majority ?

I reproduce this bug from the reading an csv file of 25k lines, whereas about 25% of lines match this problem ..

I try to use tableschema for an automatic insertion of csv files into a database, with auto-matching types on remote database, without human interactions, so that kind of problem is really problematic, i met it on 2 of the 4 csv files I tested.

Hope you can help

**Comment by akariv** [comment]
@Transrian See issue #210 which is similar to yours, and #211 which fixes this problem.

**Comment by Transrian** [comment]
@akariv Thanks, I gave it a try, seems to correct all problems concerning data types !

Got one more problem, I probably should open a new ticket ?

If you open a file like that, and change the infer limit (with the code gave earlier)

https://gist.github.com/Transrian/971cf7bc9ab0c4558cecca2fd6b4876d#file-test-csv

> table.infer(limit=102)

There still is a problem (probably not related, perhaps with the infer limit itself ?)

```bash
python3 test2.py 
| name   | type    | format   |
|:-------|:--------|:---------|
| a      | integer | default  |
| b      | integer | default  |
[CastError('Field "b" can\'t cast value "1.1" for type "integer" with format "default"',)]
```
Still thanks you very much!

**Comment by roll** [comment]
@Transrian 
Thanks for the contribution. Here - https://github.com/frictionlessdata/tableschema-py/pull/211#issuecomment-392685940 - I added some background on the topic.

**Comment by roll** [comment]
@Transrian (cc and thanks @akariv)
It has been released as `v1.1.0` with the confidence flag to solve this issue:
https://github.com/frictionlessdata/tableschema-py#tableinferlimit100-confidence075

**Comment by Transrian** [comment]
Thanks a lot, worked perfectly !
